File: app/src/main/java/com/konopko/stocktest/ActivityAskTickers.java
package com.konopko.stocktest;

import android.os.Bundle;
import android.text.InputFilter;
import android.text.Spanned;
import android.view.View;
import android.widget.Toast;

import androidx.lifecycle.SavedStateViewModelFactory;
import androidx.lifecycle.ViewModelProvider;

import com.konopko.stocktest.databinding.ActivityAskTickersBinding;

public class ActivityAskTickers extends ActivityBase<ActivityAskTickersBinding> {

    private final int TAGS_MAX_LENGTH = 30;
    private ViewModelAskTickers viewModel;

    @Override
    protected int getLayoutId() {
        return R.layout.activity_ask_tickers;
    }

    @Override
    protected String getPageTitle() {
        return getString(R.string.app_name);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding.setHandler(this);

        viewModel = new ViewModelProvider(this, new SavedStateViewModelFactory(getApplication(), this)).get(ViewModelAskTickers.class);
        viewModel.getDataNext().observe(this, open -> {
            if (open)
                openActivityList();
            else
                Toast.makeText(this, R.string.tickers_list_error, Toast.LENGTH_SHORT).show();
        });

        binding.tagsEditText.setMaxLines(2);
        binding.tagsEditText.setFilters(new InputFilter[] {
                new InputFilter() {
                    @Override
                    public CharSequence filter(CharSequence cs, int start, int end, Spanned spanned, int dStart, int dEnd) {
                        if(cs.equals(" ")) // ignore space
                            return "";
                        return cs;
                    }
                },
                new InputFilter.AllCaps(), // set all caps
                new InputFilter.LengthFilter(TAGS_MAX_LENGTH)
        });

    }

    public void onClickNext(View view){
        viewModel.next(binding.tagsEditText.getTags());
    }

    private void openActivityList(){
        ActivityTickerList.open(this);
        clearTickers();
    }

    private void clearTickers(){
        binding.tagsEditText.setTags(null);
    }
}

File: app/src/main/java/com/konopko/stocktest/ActivityAskTickers.kt
package com.konopko.stocktest

import android.os.Bundle
import android.text.InputFilter
import android.text.InputFilter.AllCaps
import android.text.InputFilter.LengthFilter
import android.view.View
import android.widget.Toast
import androidx.lifecycle.Observer
import androidx.lifecycle.SavedStateViewModelFactory
import androidx.lifecycle.ViewModelProvider
import com.konopko.stocktest.databinding.ActivityAskTickersBinding

class ActivityAskTickers : ActivityBase<ActivityAskTickersBinding>() {

    private val TAGS_MAX_LENGTH = 30
    private lateinit var viewModel: ViewModelAskTickers

    override fun getLayoutId(): Int {
        return R.layout.activity_ask_tickers
    }

    override fun getPageTitle(): String? {
        return getString(R.string.app_name)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding.handler = this

        binding.tagsEditText.maxLines = 2
        binding.tagsEditText.filters = arrayOf(
                InputFilter { cs, start, end, spanned, dStart, dEnd -> if (cs == " ") "" else cs },
                AllCaps(),  // set all caps
                LengthFilter(TAGS_MAX_LENGTH)
        )

        viewModel = ViewModelProvider(this, SavedStateViewModelFactory(application, this)).get(ViewModelAskTickers::class.java)

        viewModel.dataNext.observe(this, Observer {open: Boolean ->
                if (open)
                    openActivityList()
                else
                    Toast.makeText(this, R.string.tickers_list_error, Toast.LENGTH_SHORT).show()
        })

    }

    fun onClickNext(view: View) = viewModel.next(binding.tagsEditText.getTags())

    private fun openActivityList() {
        ActivityTickerList.open(this)
        clearTickers()
    }

    private fun clearTickers() {
        binding.tagsEditText.setTags("")
    }
}
